### ISSUE TITLE
dune fmt should not fail to format when depopt isn't available / `and` isn't lazy

### ISSUE THREAD

**Issue by ElectreAAS** [ocaml/dune#9943]
## Expected Behavior
The command `dune fmt` should work correctly and format ml files, even if the dune files state that the files shouldn't be compiled.

## Actual Behavior

With this in the dune file

```clojure
(test
 (name foo)
 (enabled_if
  (and %{lib-available:nope} (= %{version:nope} 9.9.9))))
```
we get this error:
```
File "dune", line 4, characters 32-47:
4 |   (and %{lib-available:nope} (= %{version:nope} 9.9.9))))
                                    ^^^^^^^^^^^^^^^
Error: Package "nope" doesn't exist in the current project and isn't
installed either.
```

## Special Note

The `and` expression isn't evaluated lazily, this is possibly a bug too.

**Comment by Alizter** [comment]
The and expression is currently evaluated in parallel. I think that this is less intuitive for users and I believe that implementing short-circuiting would be a better user experience overall.

**Comment by ElectreAAS** [comment]
The fmt part is better explained and adressed here #11037
the `and` bug was solved in #15291
closing in favour of the former
